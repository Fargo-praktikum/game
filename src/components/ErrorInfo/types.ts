export interface ErrorInfoProps {
    errorName: string;
    message: string;
}
