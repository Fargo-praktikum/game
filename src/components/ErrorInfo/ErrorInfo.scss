@import "../../styles/variables";

.error-info {
    align-items: center;
    display: flex;
    flex-direction: column;

    &__head {
        font-size: 40px;
        line-height: 40px;
        font-weight: $font-weight-semi-bold;
    }

    &__message {
        font-size: 20px;
        line-height: 20px;
        margin-top: 20px;
        font-weight: $font-weight-semi-bold;
    }

    &__back-link {
        margin-top: 69px;
        color: $main-blue;
        font-size: 11px;
        text-align: center;
        text-decoration: none;

        &:hover {
            color: $main-red;
        }
    }
}
