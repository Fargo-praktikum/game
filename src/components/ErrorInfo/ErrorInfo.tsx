import React from "react";
import { Link } from "react-router-dom";
import { ErrorInfoProps } from "./types";

import "./ErrorInfo.scss";

export const ErrorInfo = ({ errorName, message }: ErrorInfoProps): JSX.Element => {
    return (
        <div className="error-info">
            <h1 className="error-info__head">
                {errorName}
            </h1>
            <div className="error-info__message">
                {message}
            </div>
            <Link to="/" className="error-info__back-link">
                На главную
            </Link>
        </div>
    );
};
