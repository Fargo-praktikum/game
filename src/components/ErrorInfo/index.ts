import { ErrorInfo } from "./ErrorInfo";

export { ErrorInfo };
