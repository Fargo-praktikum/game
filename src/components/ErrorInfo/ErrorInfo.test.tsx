import React from "react";

import renderer from "react-test-renderer";
import { ErrorInfo } from "./ErrorInfo";
// здесь используем StaticRouter вместо BrowserRouter, иначе будет ошибка
import { StaticRouter } from "react-router-dom";

describe("Проверка ErrorInfo", () => {
    it("отрисовка", () => {
        const tree = renderer.create(
            <StaticRouter >
                <ErrorInfo errorName="testName" message="testMessage"/>
            </StaticRouter >
        ).toJSON();
        expect(tree).toMatchSnapshot();
    });
});



