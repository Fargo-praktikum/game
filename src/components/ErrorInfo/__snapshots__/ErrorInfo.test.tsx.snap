// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Проверка ErrorInfo отрисовка 1`] = `
<div
  className="error-info"
>
  <h1
    className="error-info__head"
  >
    testName
  </h1>
  <div
    className="error-info__message"
  >
    testMessage
  </div>
  <a
    className="error-info__back-link"
    href="/"
    onClick={[Function]}
  >
    На главную
  </a>
</div>
`;
